crates/ruff_linter/src/rules/flake8_implicit_str_concat/rules/implicit.rs

```diff
@@ -101,28 +101,37 @@ pub(crate) fn implicit(
     for ((a_tok, a_range), (b_tok, b_range)) in tokens
         .iter()
         .flatten()
-        .filter(|(tok, _)| {
-            !tok.is_comment()
-                && (settings.flake8_implicit_str_concat.allow_multiline
-                    || !tok.is_non_logical_newline())
+        .filter_map(|(tok, range)| {
+            // Ignore comments.
+            if tok.is_comment() {
+                return None;
+            }
+
+            // Ignore non-localized newlines.
+            if !settings.flake8_implicit_str_concat.allow_multiline && tok.is_non_logical_newline()
+            {
+                return None;
+            }
+
+            Some((TokenKind::from(tok), *range))
```

Comment:
Is the implicit clone because of tuple_windows? The clone shouldn't e a problem here as far as I understand because the iterator items are references and not owned Tokens

Reply:
Ah right I was misreading. Was just trying to understand why these rules are SO expensive.

Reply:
(Meant this to be a draft, just wanted the benchmarks.)